### ISSUE TITLE
Revise stops detecting changes when using CIFS and NFS.

### ISSUE THREAD

**Issue by korsbo** [timholy/Revise.jl#352]
Hi,

I have for a long time had issues with Revise.jl stopping to detect changes after a few modifications of my source code. I finally had enough and spent some time hunting down the problem. 

**Tldr:** I don't think that revise works well with cifs-mounted filesystems and hardly at all with NFS-mounted systems. 

I would fully understand if this is a low-priority issue but since I already did some work to hunt it down, I might as well report it. 

### The problem

Here is what I did (which is not to say that every step is necessary for reproduction):

1.  Generate a package using `] generate MyModule`.
2.  Activate the `MyModule` environment by 
```julia
; cd MyModule
] activate .
```
3.  Edit `src/MyModule.jl` to export a function `f`:
```julia
module MyModule

greet() = print("Hello World!")

f() = "hello"
export greet, f

end # module
```
4. run `using MyModule` in my REPL (which has already loaded Revise.jl).
5. Execute `f()`, which returned `"hello"`
6. Modify `f() = :hello` in `MyModule` and re-run `f()` in the REPL. This returned `:hello` as expected.
7. Modify `f() = :hi` in `MyModule` and re-run `f()` in the REPL. This returned `:hello` which is not expected.

Even if I manually run `revise()` from this point, it will not detect any changes.

The only solution I have found to start tracking again is to restart the REPL. 

### Complicating factors

The file-system of my computer has some complications: 
-  My `~/home/` is not local on my machine but rather mounted using NFS (version 3) from a Ubuntu 14.04 server.  My local machine is running Ubuntu 16.04. 
-  The folder where I keep all of my personal projects (including MyModule) is on a mounted using cifs on a windows server. 

Repeating the above steps on the local hard drive, the mounted `/home/` (nfs) and the mounted `.../projects/` (cifs) all give different results.

-  Local seems to work fine (but is not backed up so I don't want to work from there).
-  `/home/` does not seem to work at all. Neither for `using MyModule` nor when using `includet()`.
-  `.../projects/` works for a few revisions but then stops. Manually calling `revise()` or `revise(MyModule)` does not fix the issue. 

```julia
(v1.2) pkg> test Revise
```
passes fine which is a bit surprising since my `~/.julia/` folder is on the NFS server.

Setting `JULIA_REVISE_POLL=1` before starting the REPL does no good. It also does not slow down the revision on the local drive which I would have expected.

I am aware of the `inotify` issue with NFS, so that issue may not be surprising, but the cifs issue is news to me. 

My issues aside, this is a game-changing package; Thanks!

**Comment by timholy** [comment]
Thanks for the thorough report. Just to verify, `Revise.polling_files[]` returns `true` for you, right?

The only thing I can really recommend is to turn on [logging](https://timholy.github.io/Revise.jl/stable/debugging/). I just pushed a branch called `teh/more_logging` which might be handy. (It breaks some of the tests, which is why I am not immediately planning to submit it as a PR.) If you decide to investigate further, please let me know what you find out!

**Comment by korsbo** [comment]
> Thanks for the thorough report. Just to verify, `Revise.polling_files[]` returns `true` for you, right?
No, it returns false.

![image](https://user-images.githubusercontent.com/16306680/64182014-6ac4ad00-ce5f-11e9-9da2-5ab571024c0d.png)

**Comment by timholy** [comment]
OK, then I'm guessing you're not setting it properly. Maybe
```sh
export JULIA_REVISE_POLL=1
julia --project
```
or
```sh
JULIA_REVISE_POLL=1 julia --project
```

For example:
```julia
$ JULIA_REVISE_POLL=1 julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.2.1-pre.0 (2019-08-20)
 _/ |\__'_|_|_|\__'_|  |  release-1.2/93929550b6* (fork: 138 commits, 145 days)
|__/                   |

julia> Revise.polling_files[]
true
```

Best would be to set this in your `.bashrc` or equivalent.

**Comment by korsbo** [comment]
That did indeed help with this particular problem. `Revise.polling_files[]` returns `false` after setting  `export JULIA_REVISE_POLL=1` in the shell.

**Comment by korsbo** [comment]
At a fist glance, this seems to help for both NFS and CIFS. It does introduce the revision delay, but that is still much better than it was before.

**Comment by timholy** [comment]
I'll close this, but if you encounter more specific issues please file another issue.

**Comment by korsbo** [comment]
Thanks!
